graphene_django/fields.py

```diff
@@ -129,25 +132,29 @@ def resolve_queryset(cls, connection, queryset, info, args):
     @classmethod
     def resolve_connection(cls, connection, args, iterable, max_limit=None):
         iterable = maybe_queryset(iterable)
-        # When slicing from the end, need to retrieve the iterable length.
-        if args.get("last"):
-            max_limit = None
+
         if isinstance(iterable, QuerySet):
-            _len = max_limit or iterable.count()
+            list_length = iterable.count()
+            list_slice_length = max_limit or list_length
         else:
-            _len = max_limit or len(iterable)
+            list_length = len(iterable)
+            list_slice_length = max_limit or list_length
+
+        after = get_offset_with_default(args.get("after"), -1) + 1
+        list_slice_length += after
```

Comment:
Without this, `graphql-relay` v2 slices an empty list here:

https://github.com/graphql-python/graphql-relay-py/blob/54fa42716e587c21e1fd8e6fc7d39cb32db28039/graphql_relay/connection/arrayconnection.py#L79-L82

Might be fixed in 3.0, but I didn't try that since it might be a rabbit hole.